lib/mix/lib/mix/tasks/deps.compile.ex

```diff
@@ -33,14 +33,18 @@ defmodule Mix.Tasks.Deps.Compile do
   import Mix.Dep, only: [loaded: 1, available?: 1, loaded_by_name: 2,
                          make?: 1, mix?: 1]
 
-  @switches [include_children: :boolean]
+  @switches [include_children: :boolean, force: :boolean]
 
   @spec run(OptionParser.argv) :: :ok
   def run(args) do
     unless "--no-archives-check" in args do
       Mix.Task.run "archive.check", args
     end
 
+    if "--force" in args do
```

Comment:
I noticed the arguments aren't available in the ```compile``` look (or did I miss something?).

Would it make sense to move the check inside the ```case OptionParser.parse(args, switches: @switches) do``` and pass a flag to ```compile``` when ```force``` is set?  
Although, it feels compile and clean are two different actions.

I hope you don't mind, I'm following these changes to learn a bit more 😄